Sigmoid update for rocm rel 7.2

mpanoop opened the pull request:
## Motivation

Previous PR at https://github.com/ROCm/rocm-libraries/pull/2274 introduced the Sigmoid activation and epilogue functionality but since cublaslt is not providing a sigmoid function, this feature needs to be marked as hipblaslt extension. So made sure to add _EXT suffix to the enum values introduced for both Sigmoid and Sigmoid + bias epolgues (both in hipblaslt and rocblaslt abstractions). This PR is a cherrypick of the commits from https://github.com/ROCm/rocm-libraries/pull/2300.

## Technical Details

The technical details are same as what is furnished in PR https://github.com/ROCm/rocm-libraries/pull/2274 except the enum value change.

## Test Plan

Similar to the one published in the previous PR.

## Test Result

All the tests in hipblaslt-test successfully passed.

## Submission Checklist

- [ ] Look over the contributing guidelines at https://github.com/ROCm/ROCm/blob/develop/CONTRIBUTING.md#pull-requests.

mpanoop committed:
Updated Sigmoid Enum Value as extensions, added new BIAS + SIGMOID epilogue

Commit:
Fixed the formatting issues in PR#2300

CurtisFu1002 committed:
Updated the changelog for sigmoid activation feature

CurtisFu1002 committed:
Updated the changelog for sigmoid activation feature

CurtisFu1002 committed:
Updated the changelog for sigmoid activation feature

Commit:
Updated the CHANGELOG.md for Sigmoid Function Support in rocm-rel-7.2

amd-jnovotny commented on projects/hipblaslt/CHANGELOG.md:
```diff
@@ -6,7 +6,8 @@ Full documentation for hipBLASLt is available at [rocm.docs.amd.com/projects/hip
 
-* Support for the 'BF16' data type for gfx90a.
+* Support for 'BF16' input with 'FP32' output data type for gfx90a.
+* Support for Sigmoid Activation function.
```
If this is on the 7.2.1 cherry-pick list, then it needs to go into the 7.2.1 changelog section and the version number needs to be bumped.

jichangjichang replied:
@amd-jnovotny https://github.com/ROCm/rocm-libraries/pull/2274 is in 7.2 to enable Sigmoid Activation function.

amd-jnovotny reviewed (changes requested):
need to clarify the release and changelog info.

mpanoop commented:
@amd-jnovotny , Have updated the changelog for this feature.

amd-jnovotny commented on projects/hipblaslt/CHANGELOG.md:
```diff
@@ -2,12 +2,19 @@
+
+### Changed
+
+* Enumeration value update to the Sigmoid Activation Function feature.
```
```suggestion
* Enumeration value update for the Sigmoid Activation Function feature.
```

amd-jnovotny reviewed (approved):
Looks good. Just one minor suggestion.

mpanoop committed:
Update projects/hipblaslt/CHANGELOG.md

Co-authored-by: Jeffrey Novotny <jnovotny@amd.com>

mpanoop committed:
Update projects/hipblaslt/CHANGELOG.md

Co-authored-by: Jeffrey Novotny <jnovotny@amd.com>

darrenhsieh-amd committed:
Cherry-pick to release/rocm-rel-7.2: [hipblaslt] Add 305 problem sizes in equality for gfx942 80cu. (#2861)

## Motivation

Add equality sizes for gfx942 80cu.

<!-- Explain the purpose of this PR and the goals it aims to achieve.
-->

## Technical Details

<!-- Explain the changes along with any relevant GitHub links. -->

## Test Plan

hipblaslt-test

<!-- Explain any relevant testing done to verify this PR. -->

## Test Result

```
[----------] Global test environment tear-down
[==========] 62972 tests from 11 test suites ran. (3147144 ms total)
[  PASSED  ] 62972 tests.
```

<!-- Briefly summarize test outcomes. -->

## Submission Checklist

- [x] Look over the contributing guidelines at
https://github.com/ROCm/ROCm/blob/develop/CONTRIBUTING.md#pull-requests.

yclamd committed:
[hipblaslt] Fix workspace minimum bug in rocblaslt_mat (#3387)

Motivation
Avoid potential overload/macros from using an unqualified min(). Use
std::min<size_t> for correct and consistent workspace clamping.

Technical Details
```
if (algo) {
    workspaceSizeInBytes = std::min<size_t>(workspaceSizeInBytes, algo->max_workspace_bytes);
}
```
No functional behavior change. Only ensures type-safe clamp.

Before
[Info][rocblaslt_matmul_impl] Before min: workspaceSizeInBytes=3164160,
algo->max_workspace_bytes=4294967296
[Info][rocblaslt_matmul_impl] After min: workspaceSizeInBytes=0
[Info][runContractionProblem] Input workspace size 0 is less than the
required workspace size 3164160

After
[Info][rocblaslt_matmul_impl] Before min: workspaceSizeInBytes=3164160,
algo->max_workspace_bytes=4294967296
[Info][rocblaslt_matmul_impl] After min: workspaceSizeInBytes=3164160
[Info][runContractionProblem] problem.workspaceSize=3164160 (correct)

This shows that the incorrect min() usage resulted in workspace size
becoming 0, while std::min<size_t> preserves the correct intended size.

Test Plan
Build hipBLASLt and run existing matmul tests.

Test Result
Build and tests pass. No correctness or performance changes observed.
Workspace is now clamped correctly.

## Motivation

<!-- Explain the purpose of this PR and the goals it aims to achieve.
-->

## Technical Details

<!-- Explain the changes along with any relevant GitHub links. -->

## Test Plan

<!-- Explain any relevant testing done to verify this PR. -->

## Test Result

<!-- Briefly summarize test outcomes. -->

## Submission Checklist

- [ ] Look over the contributing guidelines at
https://github.com/ROCm/ROCm/blob/develop/CONTRIBUTING.md#pull-requests.

jharryma committed:
[CI] Use fractional runners for linux tests (#3609)

## Motivation

On this rocm-libraries PR To the 7.2 branch
(https://github.com/ROCm/rocm-libraries/pull/3403) I was seeing an issue
where linux tests are timing out waiting for a runner:

https://github.com/ROCm/rocm-libraries/actions/runs/20318068285/attempts/1?pr=3403

https://github.com/ROCm/rocm-libraries/actions/runs/20318068285/attempts/2?pr=3403
  
It looks like the runner group was replaced with the fractional runner

## Technical Details

Updating the runner to use the frac runner instead. 

## Test Plan

CI

## Test Result


## Submission Checklist

- [x] Look over the contributing guidelines at
https://github.com/ROCm/ROCm/blob/develop/CONTRIBUTING.md#pull-requests.

amd-jnovotny reviewed (changes requested):
The version bump to 1.2.1 was merged with a different PR into 7.2.0, so this PR needs a new version bump (likely to 1.2.2) now before it can be merged.

Commit:
[hipblasLT] SK use Tree reduction if < 2 split (#3403)

If origami gives us Parallel reduction and sk.grid/tiles==1 then we will
launch post kernel call, this PR sets reduction strategy to Tree
reduction in this case.

After calling getSKReduction, this does a check to see if sk.grid/tiles
< 2, if so we ensure the reduction type is Tree reduction. Small change
to origami intended to keep `select_grid` behaviour the same.

hipblaslt-test ran locally and passed, wait for CI, need to verify
downstream issue is resolved.

hipblaslt-test passed, awaiting other results.

- [ ] Look over the contributing guidelines at
https://github.com/ROCm/ROCm/blob/develop/CONTRIBUTING.md#pull-requests.

---------

Co-authored-by: Jeffrey Novotny <jnovotny@amd.com>

mpanoop committed:
Merge branch 'release/rocm-rel-7.2' into Sigmoid_Update_For_rocm_rel_7.2

Commit:
Updated the CHANGELOG with a hipblaslt version bump

Commit:
Updated the hipblaslt version in CMake file

Commit:
Updated the CHANGELOG with some corrections on features/bug fixes
